[Joy] Add principles page (#32648)
mui/material-ui: docs/data/joy/getting-started/principles/principles.md

# Principles

> <p class="description">Learn about the principles that helped shape Joy.</p>

Reviewer: ```suggestion
<p class="description">Learn about the principles that have helped shape Joy.</p>
```

## Keep it essential

> Joy should work with the least amount of effort possible. We're striving for the essential only, either on component API or design (look & feel), components should have only what they're meant to do. The definition of essentiality will be drawn from MUI's experience through the years developing component libraries like this one as well as from benchmarks of modern API and design guidelines—especially when it comes to developing web apps.

Reviewer: ```suggestion
Joy should work with the least amount of effort possible. We're striving for the essential only, both in the component API and design (look & feel). Components should have only what they need to do the job. What is considered essential will be drawn from MUI's experience over the years developing component libraries like this one, as well as from benchmarks of modern API and design guidelines—especially when it comes to developing web apps.
```

## Looks great out-of-the-box

> Joy needs to be simple yet characteristic. Visual attributes such as scale, size, and density should be consistent across all the components so they live together nicely. We look to spark delight with simplicity and attention to detail. You should feel like you're starting already looking awesome.

Reviewer: ```suggestion
Joy needs to be simple yet characteristic. Visual attributes such as scale, size, and density should be consistent across all of the components so they live together nicely. We aim to spark delight with simplicity and attention to detail. You should feel like your UI looks great from the start.
```

## Encourage creativity

> We're aiming for Joy to be entirely customizable, seen as a starting point. This is meant to encourage you to extend, change, and revamp how Joy looks. Be creative by making it your own.

Reviewer: ```suggestion
We're aiming for Joy to be entirely customizable, and seen as a great starting point. This is meant to encourage you to extend, change, and revamp how Joy looks. Be creative by making it your own.
```

## Focus on developer experience

> A great developer experience is not only the code quality we ship but also how clear the documentation writing is and how available learning resources are for developers. We hope that bundling it all together sparks joy.

Reviewer: ```suggestion
A great developer experience is not only the quality of the code we ship but also how clear the documentation is, and what learning resources are available for developers. We hope that bundling it all together brings joy.
```
